### DISCUSSION TITLE
Using the raytracer with GPU and tensor flow with CPU

### DISCUSSION THREAD

**Discussion by itsghaniz** [NVlabs/sionna#618] [General]
Hi, I am using windows 10 for my simulations. As windows doesn't support new versions of tf with GPU, I was wondering if it is possible to use the raytracing with GPU when the tensorflow cpu is installed.
Thanks in advance

**Comment by merlinND** [comment]
Hello @itsghaniz,

This is not officially supported. I have not tried this myself, but maybe you could:
1. Set the Mitsuba variant to a GPU backend, e.g. `mi.set_variant("cuda_ad_rgb")`
2. Let TensorFlow perform computations on the CPU (if it does not see any GPUs available), or force it with a `with tf.device("cpu"):` scope

But this has never been tested, so I would be surprised if something goes wrong.

**Reply by itsghaniz** [reply]
Hi, really appreciate the reply. I dis as you suggested. Although the code is working, but I am not able to see any use of the GPU. 
Thanks

**Reply by merlinND** [reply]
Hello @itsghaniz,

I'm not sure what else to suggest since this is not supported or really expected to work.
We can leave this discussion open in case someone figures it out.

**Comment by wang-chenlong** [comment]
I would like to add a bit to this issue. Initially, I also wanted to use Sionna on Windows, but TensorFlow versions greater than 2.10 do not support GPU on Windows, which is incompatible with the recommended TensorFlow versions (2.13-2.15) for Sionna.

To use a newer TensorFlow version, WSL would be required. However, as of now, WSL does not support using OptiX, which is essential for Sionna.

As a result, I ultimately configured my computer with an Ubuntu system, and now everything is working fine : )

**Comment by merlinND** [comment]
For users trying to use Sionna on the GPU on Windows / WSL2, please see this comment: https://github.com/NVlabs/sionna/issues/593#issuecomment-2505551708

**Reply by jmoraispk** [reply]
**It worked for me!**
